--- screenfetch-dev	2019-11-12 21:59:19.000000000 +0000
+++ screenfetch-dev	2022-12-30 21:06:29.560632384 +0000
@@ -364,7 +364,7 @@
 Fux, Gentoo, gNewSense, Guix System, Hyperbola GNU/Linux-libre, januslinux, Jiyuu Linux, Kali Linux, KaOS, KDE neon, Kogaion, Korora, \
 LinuxDeepin, Linux Mint, LMDE, Logos, Mageia, Mandriva/Mandrake, Manjaro, Mer, Netrunner, NixOS, OBRevenge, openSUSE, \
 OS Elbrus, Oracle Linux, Parabola GNU/Linux-libre, Pardus, Parrot Security, PCLinuxOS, PeppermintOS, Proxmox VE, PureOS, Qubes OS, \
-Raspbian, Red Hat Enterprise Linux, ROSA, Sabayon, SailfishOS, Scientific Linux, Siduction, Slackware, Solus, Source Mage GNU/Linux, \
+Raspbian, Red Hat Enterprise Linux, ROSA, Sabayon, SailfishOS, Scientific Linux, Siduction, Slackware, SliTaz, Solus, Source Mage GNU/Linux, \
 SparkyLinux, SteamOS, SUSE Linux Enterprise, SwagArch, TinyCore, Trisquel, Ubuntu, Viperr, Void and Zorin OS and EndeavourOS"
 
 supported_other="Dragonfly/Free/Open/Net BSD, Haiku, Mac OS X, Windows+Cygwin and Windows+MSYS2."
@@ -1104,6 +1104,10 @@
 							distro = "Solus"
 							exit
 						}
+						else if ($0 ~ /"SliTaz"/) {
+							distro = "SliTaz"
+							exit
+						}
 						else if ($0 ~ /"ALDOS"/) {
 							distro = "ALDOS"
 							exit
@@ -1233,6 +1237,7 @@
 		scientific*) distro="Scientific Linux" ;;
 		siduction) distro="Siduction" ;;
 		slackware) distro="Slackware" ;;
+		slitaz) distro="SliTaz" ;;
 		smgl|source*mage|source*mage*gnu*linux) distro="Source Mage GNU/Linux" ;;
 		solus) distro="Solus" ;;
 		sparky|sparky*linux) distro="SparkyLinux" ;;
@@ -1358,6 +1363,8 @@
 			pkgs=$(dpkg -l | grep -c '^i') ;;
 		'Slackware')
 			pkgs=$(ls -1 /var/log/packages | wc -l) ;;
+		'SliTaz')
+			pkgs=$(tazpkg list | wc -l) ;;		
 		'Gentoo'|'Sabayon'|'Funtoo'|'Kogaion')
 			pkgs=$(ls -d /var/db/pkg/*/* | wc -l) ;;
 		'NixOS')
@@ -4130,6 +4137,33 @@
 "${c1}                 ::::::::::::                 %s")
 		;;
 
+		"#SliTaz")
+			if [[ "$no_color" != "1" ]]; then
+				c1=$(getColor 'light grey') # Light Grey
+			fi
+			if [ -n "${my_lcolor}" ]; then c1="${my_lcolor}"; fi
+			startline="0"
+			fulloutput=(                                 
+"${c1}              #          #  	    %s"       
+"${c1}             #           ##    #   %s"  
+"${c1}      ##   ##             ###  ##  %s" 
+"${c1}     ##  ###               ###  ## %s"
+"${c1}    ##  ###                 ##  ## %s"
+"${c1}   ### ####     #####      #### ###%s"
+"${c1}  ####  #####  #######  #####  ####%s"
+"${c1}   ############################    %s"
+"${c1}       ######################      %s"   
+"${c1} ################################# %s"
+"${c1} ##    ################ ####    ## %s"
+"${c1} #  ###    ############    ###  ## %s"
+"${c1} #  ##     ###########     ###  #  %s"
+"${c1} #   #     ###########     #       %s"    
+"${c1}            ########               %s"  
+"			           %s"
+"                                  %s"
+"                                  %s")  
+		;;
+
 		"ROSA")
 			if [[ "$no_color" != "1" ]]; then
 				c1=$(getColor 'rosa_blue') # special blue color from ROSA
@@ -5850,7 +5884,7 @@
 				if [[ "$no_color" != "1" ]]; then
 					c1=$(getColor 'white') # White
 					c2=$(getColor 'dark grey') # Light Gray
-					c3=$(getColor 'yellow') # Light Yellow
+					c3=$(getColor 'light red') # Light Red
 				fi
 				if [ -n "${my_lcolor}" ]; then c1="${my_lcolor}"; c2="${my_lcolor}"; c3="${my_lcolor}"; fi
 				startline="0"
@@ -6080,7 +6114,7 @@
 			labelcolor=$(getColor 'light green')
 		;;
 		"Ubuntu"|"FreeBSD"|"FreeBSD - Old"|"Debian"|"Raspbian"|"BSD"|"Red Hat Enterprise Linux"|"Oracle Linux"| \
-		"Peppermint"|"Cygwin"|"Msys"|"Fuduntu"|"Scientific Linux"|"DragonFlyBSD"|"BackTrack Linux"|"Red Star OS"| \
+		"Peppermint"|"Cygwin"|"Msys"|"Fuduntu"|"Scientific Linux"|"SliTaz"|"DragonFlyBSD"|"BackTrack Linux"|"Red Star OS"| \
 		"SparkyLinux"|"OBRevenge"|"Source Mage GNU/Linux")
 			labelcolor=$(getColor 'light red')
 		;;
